### PR TITLE
Sync types with github.com/shutter-network/go-ethereum

### PR DESCRIPTION
**Pull request by ezdac**
Using the `shtypecopy` [tool](https://github.com/shutter-network/txtypes/tree/main/shtypecopy) this PR syncs all files specified in `rule.shtypecopy` from the `shutter-types` branch of our go-ethereum fork to this fork:

https://github.com/shutter-network/cannon/blob/55eac995de09a3856e05b68b4eadf47e732c2bbd/minigeth/core/types/rules.shtypecopy#L1-L14

There are additional files written out from the tool that describe the HEAD commit used to sync the files, as well as the last 
common ancestor with the upstream go-ethereum's `master` branch (so the underlying geth version).

The `shtypecopy` command used was:

```bash
cd cannon/minigeth
shtypecopy --in core/types/ --out core/types --rules core/types/rules.shtypecopy
```

### CODE REVIEW

**Review by jannikluhn — APPROVED** [review]
LGTM, just two naming suggestions:

- `BatchContextTx` -> `BatchTx`, since it now represents the full batch. Before it was only the context.
- `ShutterTXs` and `PlainTextTXs`: Not sure about this, but all transaction types don't capitalize the `X` (e.g., `AccessListTx`), so maybe better to call these fields `ShutterTxs` and `PlainTextTxs`

**Comment by ezdac** [comment]
Will change that.
By the way, do we even need the separation of the two?
Couldn't we just have *one* field where we put both types?

File: minigeth/core/types/batch_tx.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	Transactions  [][]byte
+
+	// Signature values
+	V *big.Int `json:"v" gencodec:"required"`
+	R *big.Int `json:"r" gencodec:"required"`
+	S *big.Int `json:"s" gencodec:"required"`
+}
+
+// copy creates a deep copy of the transaction data and initializes all fields.
+func (tx *BatchTx) copy() TxData {
```
Comment (jannikluhn) [inline minigeth/core/types/batch_tx.go:24]: looks like the signature fields are missing here

File: minigeth/core/types/transaction_signing.go [hunk 2]
```diff
@@ -182,6 +182,7 @@ func NewLondonSigner(chainId *big.Int) Signer {
 }
 
 func (s londonSigner) Sender(tx *Transaction) (common.Address, error) {
+	// TODO this also has to include the shutter types
```
Comment (jannikluhn) [inline minigeth/core/types/transaction_signing.go:185]: either I don't understand the comment or this is already done. `eip2930Signer` works for shutter and batch txs, no?

Comment (ezdac) [reply minigeth/core/types/transaction_signing.go:185]: Yes I'm sorry, this is a leftover

File: minigeth/core/types/batch_tx.go (hunk 1 above)
Comment (ezdac) [reply minigeth/core/types/batch_tx.go:24]: good catch

Comment (ezdac) [reply minigeth/core/types/batch_tx.go:24]: Fixed

**Review by jannikluhn — APPROVED** [review]
LGTM, can you merge?

**Pull request merged**